File: Day292/elminateMaximumNumberOfMonsters.java
class Solution {
    public int eliminateMaximum(int[] dist, int[] speed) {
       double arr[] = new double[dist.length];
       for (int i=0; i<dist.length; i++) {
           arr[i] = (double)dist[i]/ speed[i];
       }
       Arrays.sort(arr);
       for (int i=0; i<dist.length; i++) {
           if (arr[i]<=(double)i) {
               return i;
           }
       }
       return dist.length;
    }
}

File: Day292/elminateMaximumNumberOfMonsters.go
func eliminateMaximum(dist []int, speed []int) int {
    arr := make([]float64, len(dist))
    for i:=0; i<len(dist); i++ {
        arr[i] = float64(dist[i])/float64(speed[i])
    }
    sort.Float64s(arr)
    for i:=0; i<len(dist); i++ {
        if arr[i] <= float64(i) {
            return i
        }
    }
    return len(dist)
}